src/pip/_internal/configuration.py

```diff
@@ -35,6 +35,20 @@
     RawConfigParser = configparser.RawConfigParser  # Shorthand
     Kind = NewType("Kind", str)
 
+CONFIG_BASENAME = 'pip.ini' if WINDOWS else 'pip.conf'
+ENV_NAMES_IGNORED = "version", "help"
+
+# The kinds of configurations there are.
+kinds = enum(
+    USER="user",        # User Specific
+    GLOBAL="global",    # System Wide
+    SITE="site",        # [Virtual] Environment Specific
+    ENV="env",          # from PIP_CONFIG_FILE
+    ENV_VAR="env-var",  # from Environment Variables
+)
+OVERRIDE_ORDER = kinds.GLOBAL, kinds.USER, kinds.SITE, kinds.ENV, kinds.ENV_VAR
```

Comment:
Would it be possible to build this ordering into `enum`? I’m not very familiar with how the functional API works, but it’s be nicer to iterate directly on `kinds` if possible.

Reply:
I don't think it's possible, at least if `enum(**kwargs)` is used since `dict` didn't preserve order until Python 3.7.  After dropping Python 2 however we can use [the standard library](https://docs.python.org/3/library/enum.html) though.

I start to wonder if we want to start noting a list of things to be done when Python 2 support is retired.  I alone have encounter quite a few but I can't remember anything else at the moment.

Reply:
I looked at the implementation and indeed it’s not possible, the `dict` call unfortunately looses any ordering. It’s quite trivial to fix, but probably out of the scope of this PR.